### PR TITLE
Report errors if no files were found or if source-directories are empty

### PR DESCRIPTION
**Pull request by jfmengels**
Supersedes #341, by addressing the root problem.

As @lishaduck pointed out in https://github.com/jfmengels/node-elm-review/pull/341#issuecomment-2676439308

> we don't have any tests where there isn't a source dir, do we? I have a sneaking suspicion that the reason it hung was that tinyglobby <0.2.11 would've scanned the entire drive when given an absolute path, but in case I'm wrong there, we should probably make sure elm-review doesn't hang in the actual situation where there aren't any files.

Indeed, when there are no Elm files at all, `elm-review` hangs. I added a test for it, and we're now exiting with an error should that happen. I also added a specific error (and test) for when `source-directories` is empty for some reason.

I also `unique`d source-directories we're running, because there was some duplicates otherwise, and that was one additional `glob` call.

### CODE REVIEW

File: lib/elm-files.js [hunk 1]
```diff
@@ -52,6 +53,9 @@ async function getProjectFiles(options, elmSyntaxVersion) {
   const emptyDirectories = filesInDirectories.filter(
     ({files}) => files.length === 0
   );
+  const elmFilesToRead = [
+    ...new Set(filesInDirectories.flatMap((directory) => directory.files))
+  ];
```
Comment (lishaduck) [inline lib/elm-files.js:58]: Just noticed that is just unique() inlined. Maybe I'll de-inline it someday. Probably not. :)

Comment (jfmengels) [reply lib/elm-files.js:58]: Ah right. Yeah I reached out for `unique` after writing this code. I'll make the change.

Comment (lishaduck) [reply lib/elm-files.js:58]: I was rebasing and this made a conflict, so I fixed it in #344.

Comment (jfmengels) [reply lib/elm-files.js:58]: Just saw your message. I fixed it in #354, it's on master now :sweat_smile:

Comment (lishaduck) [reply lib/elm-files.js:58]: Already re-(re-?)rebased
